adapter/internal/api/apis_impl.go

```diff
@@ -218,37 +271,102 @@ func updateAPI(name, version string, apiProject ProjectAPI, environments []strin
 	}
 }
 
-func extractAPIInformation(apiProject ProjectAPI) {
-	var apiDef map[string]interface{}
-	unmarshalErr := json.Unmarshal(apiProject.APIJsn, &apiDef)
-	if unmarshalErr != nil {
-		loggers.LoggerAPI.Errorf("Error occured while parsing api.yaml %v", unmarshalErr.Error())
+func extractAPIInformation(apiProject *ProjectAPI, apiObject config.APIJsonData) {
+	apiProject.APIType = strings.ToUpper(apiObject.Data.APIType)
+	apiProject.APILifeCycleStatus = strings.ToUpper(apiObject.Data.LifeCycleStatus)
+
+	if apiObject.Data.AuthorizationHeader != "" {
+		apiProject.AuthHeader = apiObject.Data.AuthorizationHeader
+	} else {
+		conf, _ := config.ReadConfigs()
+		apiProject.AuthHeader = conf.Security.Adapter.AuthorizationHeader
 	}
-	data := apiDef["data"].(map[string]interface{})
-	apiProject.APIType = strings.ToUpper(data[apiTypeYamlKey].(string))
-	apiProject.APILifeCycleStatus = strings.ToUpper(data[lifeCycleStatus].(string))
-
-	var securitySchemesTypes []string = nil
-	if data[securityScheme] != nil {
-		securitySchemes := data[securityScheme].([]interface{})
-		for _, scheme := range securitySchemes {
-			securitySchemesTypes = append(securitySchemesTypes, scheme.(string))
-		}
+
+	apiProject.SecurityScheme = apiObject.Data.SecurityScheme
+
+	var apiHashValue string = generateHashValue(apiObject.Data.APIName, apiObject.Data.APIName)
+
+	endpointConfig := apiObject.Data.EndpointConfig
+
+	// production Endpoints set
+	var productionEndpoint string
+	productionEndpoint = resolveEnvValueForEndpointConfig("api_" + apiHashValue + "_prod_endpoint_0")
+	if productionEndpoint == "" {
+		productionEndpoint = endpointConfig.ProductionEndpoints.Endpoint
 	}
-	apiProject.SecurityScheme = securitySchemesTypes
-	loggers.LoggerAPI.Infof("apiProject.SecurityScheme %v", apiProject.SecurityScheme)
+	apiProject.ProductionEndpoint = productionEndpoint
+	loggers.LoggerAPI.Infof("apiProject.ProductionEndpoint %v", apiProject.ProductionEndpoint)
 
-	apiProject.endpointImplementationType = data[endpointImplementationType].(string)
+	// sandbox Endpoints set
+	var sandboxEndpoint string
+	sandboxEndpoint = resolveEnvValueForEndpointConfig("api_" + apiHashValue + "_sand_endpoint_0")
+	if sandboxEndpoint == "" {
+		sandboxEndpoint = endpointConfig.SandBoxEndpoints.Endpoint
+	}
+	apiProject.SandboxEndpoint = sandboxEndpoint
+	loggers.LoggerAPI.Infof("apiProject.SandboxEndpoint %v", apiProject.SandboxEndpoint)
+
+	// production Endpoint security
+	prodEpSecurity, _ := retrieveEndPointSecurityInfo("api_"+apiHashValue,
+		endpointConfig.EndpointSecurity.Production, "prod")
+	loggers.LoggerAPI.Infof("apiProject_security %v", endpointConfig.EndpointSecurity.Production)
```

Comment:
-> Debug logs may be

Reply:
Removed with https://github.com/wso2/product-microgateway/pull/1785/commits/5e7db9dba2577979211d7713f6690403ca650326 no need debug